### ISSUE TITLE
Test failures when optional modules not present or wrong version

### ISSUE THREAD

**Issue by pmqs** [pmqs/Archive-Zip-SimpleZip#15]
https://cpantesters.org/cpan/report/bacc1c1a-ad53-11ea-a83c-75d8f4ffae69

```
From: Alceu Rodrigues de Freitas Junior
Subject: FAIL Archive-Zip-SimpleZip-0.035 5.30.2 OpenBSD
Date: 2020-06-13T08:56:11Z

This distribution has been tested as part of the CPAN Testers
project, supporting the Perl programming language.  See
http://wiki.cpantesters.org/ for more information or email
questions to cpan-testers-discuss@perl.org


--
Dear Paul Marquess,

This is a computer-generated report for Archive-Zip-SimpleZip-0.035
on perl 5.30.2, created by CPAN-Reporter-1.2018.

Thank you for uploading your work to CPAN.  However, there was a problem
testing your distribution.

If you think this report is invalid, please consult the CPAN Testers Wiki
for suggestions on how to avoid getting FAIL reports for missing library
or binary dependencies, unsupported operating systems, and so on:

http://wiki.cpantesters.org/wiki/CPANAuthorNotes

Sections of this report:

    * Tester comments
    * Program output
    * Prerequisites
    * Environment and other context

------------------------------
TESTER COMMENTS
------------------------------

Additional comments from tester:

this report is from an automated smoke testing program
and was not reviewed by a human for accuracy

------------------------------
PROGRAM OUTPUT
------------------------------

Output from '/usr/bin/make test':

PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
# 
# 
# perl                      5.030002
# Archive::Zip::SimpleZip   0.035
# Archive::Zip::SimpleUnzip 0.035
# IO::Compress::Base        2.093
# IO::Compress::Zip         2.093
# IO::Compress::Lzma        2.093
# IO::Uncompress::Base      2.093
# IO::Uncompress::Unzip     2.093
# IO::Uncompress::UnLzma    2.093
# Compress::Raw::Zlib       2.093
# Compress::Raw::Bzip2      2.089
# Compress::Raw::Lzma       2.093
# zlib                      1.2.11
# bzip2                     1.0.8, 13-Jul-2019
# lzma                      5.2.5
# 
# 
t/000prereq.t ............. ok
t/001main.t ............... ok
IO::Compress::Zip: Bzip2 not available at lib/Archive/Zip/SimpleZip.pm line 247.
# Looks like you planned 10342 tests but only ran 2588.
# Looks like your test died just after 2588.
t/010main-unzip.t ......... 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 7754/10342 subtests 
Label not found for "last SKIP" at t/Test/More.pm line 883.
# Looks like you planned 4164 tests but only ran 2755.
# Looks like your test died just after 2755.
t/020main-unzip-stream.t .. 
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1409/4164 subtests 
	(less 1 skipped subtest: 2754 okay)
t/111const-deflate.t ...... ok
t/999pod.t ................ ok
t/meta-json.t ............. ok
t/meta-yaml.t ............. ok

Test Summary Report
-------------------
t/010main-unzip.t       (Wstat: 65280 Tests: 2588 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 10342 tests but ran 2588.
t/020main-unzip-stream.t (Wstat: 65280 Tests: 2755 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 4164 tests but ran 2755.
Files=8, Tests=6273,  3 wallclock secs ( 0.30 usr  0.12 sys +  2.30 cusr  0.93 csys =  3.65 CPU)
Result: FAIL
Failed 2/8 test programs. 0/6273 subtests failed.
*** Error 255 in /mnt/cpan_build_dir/goku/Archive-Zip-SimpleZip-0.035-0 (Makefile:884 'test_dynamic')

------------------------------
PREREQUISITES
------------------------------

Prerequisite modules loaded:

requires:

    Module                 Need  Have   
    ---------------------- ----- -------
    IO::Compress::Base     2.093 2.093  
    IO::Compress::Lzma     2.093 2.093  
    IO::Uncompress::Base   2.093 2.093  
    Perl::OSType           0     1.010  

build_requires:

    Module                 Need  Have   
    ---------------------- ----- -------
    ExtUtils::MakeMaker    0     7.45_01

configure_requires:

    Module                 Need  Have   
    ---------------------- ----- -------
    ExtUtils::MakeMaker    0     7.45_01

opt_requires:

    Module                 Need  Have   
    ---------------------- ----- -------
    IO::Compress::Lzma     2.093 2.093  
    IO::Uncompress::UnLzma 2.093 2.093  


------------------------------
ENVIRONMENT AND OTHER CONTEXT
------------------------------

Environment variables:

    AUTOMATED_TESTING = 1
    DBD_MYSQL_TESTDB = test
    DBD_MYSQL_TESTHOST = localhost
    DBD_MYSQL_TESTPASSWORD = 
    DBD_MYSQL_TESTPORT = 3306
    DBD_MYSQL_TESTUSER = goku
    EXTENDED_TESTING = 1
    PATH = /home/goku/perl5/perlbrew/bin:/home/goku/perl5/bin:/home/goku/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/X11R6/bin:/usr/local/bin:/usr/local/sbin
    PERL5LIB = /home/goku/perl5/lib/perl5
    PERL5OPT = 
    PERL5_CPANPLUS_IS_RUNNING = 5206
    PERL5_CPAN_IS_RUNNING = 5206
    PERL5_CPAN_IS_RUNNING_IN_RECURSION = 58182,5206
    PERLBREW_HOME = /home/goku/.perlbrew
    PERLBREW_PATH = /home/goku/perl5/perlbrew/bin
    PERLBREW_ROOT = /home/goku/perl5/perlbrew
    PERLBREW_SHELLRC_VERSION = 0.88
    PERLBREW_VERSION = 0.88
    PERL_CR_SMOKER_CURRENT = Archive-Zip-SimpleZip-0.035
    PERL_EXTUTILS_AUTOINSTALL = --defaultdeps
    PERL_LOCAL_LIB_ROOT = /home/goku/perl5
    PERL_MB_OPT = --install_base "/home/goku/perl5"
    PERL_MM_OPT = INSTALL_BASE=/home/goku/perl5
    PERL_MM_USE_DEFAULT = 1
    PERL_USE_UNSAFE_INC = 1
    SHELL = /usr/local/bin/bash
    TERM = xterm-256color

Perl special variables (and OS-specific diagnostics, for MSWin32):

    $^X = /usr/bin/perl
    $UID/$EUID = 1001 / 1001
    $GID = 999 999 1001
    $EGID = 999 999 1001

Perl module toolchain versions installed:

    Module              Have    
    ------------------- --------
    CPAN                2.28    
    CPAN::Meta          2.150010
    Cwd                 3.78    
    ExtUtils::CBuilder  0.280234
    ExtUtils::Command   7.45_01 
    ExtUtils::Install   2.14    
    ExtUtils::MakeMaker 7.45_01 
    ExtUtils::Manifest  1.72    
    ExtUtils::ParseXS   3.40    
    File::Spec          3.78    
    JSON                4.02    
    JSON::PP            4.04    
    Module::Build       0.4231  
    Module::Signature   0.83    
    Parse::CPAN::Meta   2.150010
    Test::Harness       3.43_02 
    Test::More          1.302175
    YAML                1.30    
    YAML::Syck          n/a     
    version             0.9924  


--

Summary of my perl5 (revision 5 version 30 subversion 2) configuration:
   
  Platform:
    osname=openbsd
    osvers=6.7
    archname=amd64-openbsd
    uname='openbsd'
    config_args='-dsE -Dopenbsd_distribution=defined -Dccflags=-DNO_LOCALE_NUMERIC -DNO_LOCALE_COLLATE -Dmksymlinks'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-DNO_LOCALE_NUMERIC -DNO_LOCALE_COLLATE -fno-strict-aliasing -fno-delete-null-pointer-checks -pipe -fstack-protector-strong -D_FORTIFY_SOURCE=2 -I/usr/local/include'
    optimize='-O2'
    cppflags='-DNO_LOCALE_NUMERIC -DNO_LOCALE_COLLATE -fno-strict-aliasing -fno-delete-null-pointer-checks -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='4.2.1 Compatible OpenBSD Clang 8.0.1 (tags/RELEASE_801/final)'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags ='-Wl,-E  -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/lib /usr/lib
    libs=-lm -lc
    perllibs=-lm -lc
    libc=/usr/lib/libc.a
    so=so
    useshrplib=true
    libperl=libperl.so.20.0
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-R/usr/libdata/perl5/amd64-openbsd/CORE'
    cccdlflags='-DPIC -fPIC '
    lddlflags='-shared -fPIC  -fstack-protector-strong -L/usr/local/lib'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_CTYPE
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
  Built under openbsd
  %ENV:
    PERL5LIB="/home/goku/perl5/lib/perl5"
    PERL5OPT=""
    PERL5_CPANPLUS_IS_RUNNING="5206"
    PERL5_CPAN_IS_RUNNING="5206"
    PERL5_CPAN_IS_RUNNING_IN_RECURSION="58182,5206"
    PERLBREW_HOME="/home/goku/.perlbrew"
    PERLBREW_PATH="/home/goku/perl5/perlbrew/bin"
    PERLBREW_ROOT="/home/goku/perl5/perlbrew"
    PERLBREW_SHELLRC_VERSION="0.88"
    PERLBREW_VERSION="0.88"
    PERL_CR_SMOKER_CURRENT="Archive-Zip-SimpleZip-0.035"
    PERL_EXTUTILS_AUTOINSTALL="--defaultdeps"
    PERL_LOCAL_LIB_ROOT="/home/goku/perl5"
    PERL_MB_OPT="--install_base "/home/goku/perl5""
    PERL_MM_OPT="INSTALL_BASE=/home/goku/perl5"
    PERL_MM_USE_DEFAULT="1"
    PERL_USE_UNSAFE_INC="1"
  @INC:
    /home/goku/perl5/lib/perl5/5.30.2/amd64-openbsd
    /home/goku/perl5/lib/perl5/5.30.2
    /home/goku/perl5/lib/perl5/amd64-openbsd
    /home/goku/perl5/lib/perl5
    /usr/local/libdata/perl5/site_perl/amd64-openbsd
    /usr/local/libdata/perl5/site_perl
    /usr/libdata/perl5/amd64-openbsd
    /usr/libdata/perl5
```

**Comment by pmqs** [comment]
http://www.cpantesters.org/cpan/report/6b9ef39c-f123-11ea-bee0-766a1f24ea8f

```
Report for Archive-Zip-SimpleZip-0.038
Back
From: Slaven Rezić (SREZIC)
Subject: FAIL Archive-Zip-SimpleZip-0.038 5.32.0 GNU/Linux
Date: 2020-09-07T16:01:41Z

This distribution has been tested as part of the CPAN Testers
project, supporting the Perl programming language.  See
http://wiki.cpantesters.org/ for more information or email
questions to cpan-testers-discuss@perl.org


--
Dear Paul Marquess,

This is a computer-generated report for Archive-Zip-SimpleZip-0.038
on perl 5.32.0, created by CPAN-Reporter-1.2018.

Thank you for uploading your work to CPAN.  However, there was a problem
testing your distribution.

If you think this report is invalid, please consult the CPAN Testers Wiki
for suggestions on how to avoid getting FAIL reports for missing library
or binary dependencies, unsupported operating systems, and so on:

http://wiki.cpantesters.org/wiki/CPANAuthorNotes

Sections of this report:

    * Tester comments
    * Program output
    * Prerequisites
    * Environment and other context

------------------------------
TESTER COMMENTS
------------------------------

Additional comments from tester:

none provided

------------------------------
PROGRAM OUTPUT
------------------------------

Output from '/usr/bin/make test':

PERL_DL_NONLAZY=1 "/opt/perl-5.32.0/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t

#     Failed test (t/000prereq.t at line 58)
#          got: '2.095'
#     expected: '2.096'
# IO::Compress::Zstd version is 2.095, need 2.096

#     Failed test (t/000prereq.t at line 58)
#          got: '2.095'
#     expected: '2.096'
# IO::Uncompress::UnZstd version is 2.095, need 2.096
# 
# 
# perl                      5.032000
# Archive::Zip::SimpleZip   0.038
# Archive::Zip::SimpleUnzip 0.038
# IO::Compress::Base        2.096
# IO::Compress::Zip         2.096
# IO::Compress::Bzip2       2.096
# IO::Compress::Lzma        2.096
# IO::Compress::Xz          2.096
# IO::Compress::Zstd        2.095
# IO::Uncompress::Base      2.096
# IO::Uncompress::Unzip     2.096
# IO::Uncompress::Bunzip2   2.096
# IO::Uncompress::UnLzma    2.096
# IO::Uncompress::UnXz      2.096
# IO::Uncompress::UnZstd    2.095
# Compress::Raw::Zlib       2.096
# Compress::Raw::Bzip2      2.096
# Compress::Raw::Lzma       2.096
# Compress::Stream::Zstd    0.201
# zlib                      1.2.11
# bzip2                     1.0.8, 13-Jul-2019
# lzma                      5.2.4
# Zstandard                 1.4.3
# 
# 
# Looks like you failed 2 tests of 10.
t/000prereq.t ............. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/10 subtests 
t/001main.t ............... ok
t/010main-unzip.t ......... ok
t/020main-unzip-stream.t .. ok
t/111const-deflate.t ...... ok
t/999pod.t ................ ok
t/meta-json.t ............. ok
t/meta-yaml.t ............. ok

Test Summary Report
-------------------
t/000prereq.t           (Wstat: 512 Tests: 10 Failed: 2)
  Failed tests:  8-9
  Non-zero exit status: 2
Files=8, Tests=24348,  6 wallclock secs ( 1.53 usr  0.12 sys +  4.04 cusr  1.47 csys =  7.16 CPU)
Result: FAIL
Failed 1/8 test programs. 2/24348 subtests failed.
make: *** [Makefile:885: test_dynamic] Error 255

------------------------------
PREREQUISITES
------------------------------

Prerequisite modules loaded:

requires:

    Module               Need  Have 
    -------------------- ----- -----
    IO::Compress::Base   2.096 2.096
    IO::Compress::Lzma   2.096 2.096
    IO::Uncompress::Base 2.096 2.096
    Perl::OSType         0     1.010

build_requires:

    Module               Need  Have 
    -------------------- ----- -----
    ExtUtils::MakeMaker  0     7.46 

configure_requires:

    Module               Need  Have 
    -------------------- ----- -----
    ExtUtils::MakeMaker  0     7.46 


------------------------------
ENVIRONMENT AND OTHER CONTEXT
------------------------------

Environment variables:

    LANG = en_US.UTF-8
    PATH = /usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/home/cpansand/bin/linux-gnu:/home/cpansand/bin/sh:/home/cpansand/bin:/usr/games:/home/cpansand/devel:/home/eserte/src/srezic-misc/scripts
    PERL5LIB = 
    PERL5OPT = 
    PERL5_CPANPLUS_IS_RUNNING = 218344
    PERL5_CPAN_IS_RUNNING = 218344
    PERL5_CPAN_IS_RUNNING_IN_RECURSION = 218295,218344
    PERLDOC = -MPod::Perldoc::ToTextOverstrike
    PERL_BATCH = yes
    PERL_CANARY_STABILITY_NOPROMPT = 1
    PERL_CPAN_REPORTER_CONFIG = /var/tmp/cpansmoker-1023/2020090708/cpanreporter_000_config.ini
    PERL_EXTUTILS_AUTOINSTALL = --defaultdeps
    PERL_USE_UNSAFE_INC = 1
    SHELL = /usr/bin/zsh
    TERM = screen
    TMPDIR = /var/tmp/cpansmoker-1023/2020090708

Perl special variables (and OS-specific diagnostics, for MSWin32):

    $^X = /opt/perl-5.32.0/bin/perl
    $UID/$EUID = 1023 / 1023
    $GID = 1023 1023
    $EGID = 1023 1023

Perl module toolchain versions installed:

    Module              Have    
    ------------------- --------
    CPAN                2.27    
    CPAN::Meta          2.150010
    Cwd                 3.78    
    ExtUtils::CBuilder  0.280234
    ExtUtils::Command   7.46    
    ExtUtils::Install   2.16    
    ExtUtils::MakeMaker 7.46    
    ExtUtils::Manifest  1.72    
    ExtUtils::ParseXS   3.40    
    File::Spec          3.78    
    JSON                4.02    
    JSON::PP            4.04    
    Module::Build       0.4231  
    Module::Signature   n/a     
    Parse::CPAN::Meta   2.150010
    Test::Harness       3.42    
    Test::More          1.302177
    YAML                1.30    
    YAML::Syck          1.32    
    version             0.9924  


--

Summary of my perl5 (revision 5 version 32 subversion 0) configuration:
   
  Platform:
    osname=linux
    osvers=5.4.0-37-generic
    archname=x86_64-linux
    uname='linux cloud8.bbbike.org 5.4.0-37-generic #41-ubuntu smp wed jun 3 18:57:02 utc 2020 x86_64 x86_64 x86_64 gnulinux '
    config_args='-ds -e -Dprefix=/opt/perl-5.32.0 -Dcf_email=srezic@cpan.org'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2'
    cppflags='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='9.3.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/include/x86_64-linux-gnu /usr/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib
    libs=-lpthread -lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.31.so
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version='2.31'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
  Built under linux
  Compiled at Jun 21 2020 09:14:58
  %ENV:
    PERL5LIB=""
    PERL5OPT=""
    PERL5_CPANPLUS_IS_RUNNING="218344"
    PERL5_CPAN_IS_RUNNING="218344"
    PERL5_CPAN_IS_RUNNING_IN_RECURSION="218295,218344"
    PERLDOC="-MPod::Perldoc::ToTextOverstrike"
    PERL_BATCH="yes"
    PERL_CANARY_STABILITY_NOPROMPT="1"
    PERL_CPAN_REPORTER_CONFIG="/var/tmp/cpansmoker-1023/2020090708/cpanreporter_000_config.ini"
    PERL_EXTUTILS_AUTOINSTALL="--defaultdeps"
    PERL_USE_UNSAFE_INC="1"
  @INC:
    /opt/perl-5.32.0/lib/site_perl/5.32.0/x86_64-linux
    /opt/perl-5.32.0/lib/site_perl/5.32.0
    /opt/perl-5.32.0/lib/5.32.0/x86_64-linux
    /opt/perl-5.32.0/lib/5.32.0
    .
CPAN::Testers::Web v0.001 is powered by Mojolicious
All data and statistics are copyright © 1999-2018 CPAN Testers, and available under the Artistic License 2.0. Comments and design patches to admin@cpantesters.org.

The CPAN Testers Family:
Blog • Reports • Wiki • Statistics • PASS Matrix • Preferences • Metabase • Development • Dependencies • Matrix • Analysis • Admin

CPAN Testers, supporting the Perl programming language since 1999.
```

**Comment by pmqs** [comment]
fixed by #15
